code/modules/reagents/chemistry/machinery/reagentgrinder.dm

```diff
@@ -381,18 +407,23 @@
 	qdel(O)
 
 /obj/machinery/reagentgrinder/proc/juice()
+	var/obj/item/reagent_containers/B = null
+	if(internal)
+		B = internal
+	else
+		B = beaker
 	power_change()
 	if(stat & (NOPOWER|BROKEN))
 		return
-	if(!beaker || beaker.reagents.holder_full())
+	if(!B || B.reagents.holder_full())
 		return
 
-	playsound(src.loc, 'sound/machines/juicer.ogg', 20, 1)
+	playsound(src.loc, 'sound/machines/juicer.ogg', 20, TRUE)
 	var/offset = prob(50) ? -2 : 2
 	animate(src, pixel_x = pixel_x + offset, time = 0.2, loop = 250) // Start shaking
 	operating = TRUE
 	SStgui.update_uis(src)
-	spawn(5 SECONDS)
+	spawn((processTime - 1) SECONDS)
```

Comment:
I don't think spawn can be used

Reply:
> I don't think spawn can be used

spawn has always been used here, I only modified the time.